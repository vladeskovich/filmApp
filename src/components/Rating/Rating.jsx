import React from 'react';
import PropTypes from 'prop-types';
import BarItem from '../BarItem';
import VoteCount from '../VoteCount';
import styles from './Rating.scss';

const Rating = ({
  id,
  rating,
}) => {
  const convertedRating = Math.round(rating / 2);
  const renderStars = () => (Array.from({ length: 5 }, (v, k) => (
      <BarItem
        key={k}
        name={id}
        activeItem={convertedRating}
        id={5 - k}
      />
  )));
  return (
    <div className={styles.ratingContainer}>
      <div className={styles.ratingInStars}>
        {renderStars()}
      </div>
      <div className={styles.ratingInNumber}>
        <VoteCount rating={rating}/>
      </div>
    </div>
  );
};

Rating.propTypes = {
  data: PropTypes.string,
  className: PropTypes.string,
};

Rating.defaultProps = {};

export default Rating;
