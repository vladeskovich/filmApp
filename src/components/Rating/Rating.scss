.ratingContainer {
  display: flex;
  position: absolute;
  bottom: -60px;
  transform: rotateY(180deg);

  input {
    display: none;
  }

  label {
    display: block;
    width: 35px;
  }

  label:before {
    content: '★';
    position: relative;
    display: block;
    font-size: 35px;
    color: greenyellow;
  }

  label:after {
    content: '★';
    position: absolute;
    top: 0;
    display: block;
    font-size: 35px;
    color: #0aaee4;
    opacity: 0;
    transition: 0.5s;
  }

  label:hover ~ label:after,
  label:hover:after,
  input:checked ~ label:after {
    opacity: 1;
  }
}

