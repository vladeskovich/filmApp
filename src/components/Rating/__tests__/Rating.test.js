import React from 'react';
import Rating from '../Rating';

describe('Rating', () => {
  const props = {
    id: 5353,
    rating: 9,
  };

  it('should render Rating component with props', () => {
    const component = shallow(<Rating {...props}/>);

    expect(component).toMatchSnapshot();
  });
});
