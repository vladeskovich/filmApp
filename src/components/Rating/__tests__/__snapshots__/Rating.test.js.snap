// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating should render Rating component with props 1`] = `
<div
  className="Rating__ratingContainer_1c_xmAPs"
>
  <div
    className="Rating__ratingInStars_1ES92-ut"
  >
    <BarItem
      activeItem={5}
      id={5}
      key="0"
      name={5353}
      onChange={[Function]}
    />
    <BarItem
      activeItem={5}
      id={4}
      key="1"
      name={5353}
      onChange={[Function]}
    />
    <BarItem
      activeItem={5}
      id={3}
      key="2"
      name={5353}
      onChange={[Function]}
    />
    <BarItem
      activeItem={5}
      id={2}
      key="3"
      name={5353}
      onChange={[Function]}
    />
    <BarItem
      activeItem={5}
      id={1}
      key="4"
      name={5353}
      onChange={[Function]}
    />
  </div>
  <div
    className="Rating__ratingInNumber_3Rj40w5F"
  >
    <VoteCount
      rating={9}
    />
  </div>
</div>
`;
